### PR TITLE
Remove default empty string parameters

### PR DESCRIPTION
**Pull request by traylenator**
#### Pull Request (PR) description
For the defined types and parameters

* For `squid::cache_dir` and the parameter `options` the default is now `undef` rather than `''`
* For `squid::snmp_port` and the parameter `options` the default is now `undef` rather than `''`
* For `squid::refresh_pattern` and the parameter `options` the default is now `undef` rather than `''`
* For `squid::http_port` and the parameter `options` the default is now `undef` rather than `''`
* For `squid::http_ports` and the parameter `options` the default is now `undef` rather than `''`

This is backwards compatible for all cases except when `options` has
been explicitly set to `''` but I feel that is a non-case?

e.g.

```puppet
squid::http_port{80:
  options => '',
```
would fail.

Resolves the linter  issues for `params_empty_string_assignment`.

Extra commits include:
* modulesync 5.1.0 templates
* rubocop corrections
* Addition of docs SSLBump type

### CODE REVIEW

**Comment by traylenator** [comment]
Think I read accept' tests for 8 are currently broken somewhere....

**Comment by traylenator** [comment]
After After https://github.com/voxpupuli/puppet-squid/pull/170

File: spec/classes/init_spec.rb [hunk 1]
```diff
@@ -49,7 +49,7 @@
         it { is_expected.to contain_concat_fragment('squid_header').without_content(%r{^err_page_stylesheet}) }
       end
 
-      context 'with all parameters set' do
+      context 'with all parameters set' do # rubocop:todo RSpec/MultipleMemoizedHelpers
```
Comment (smortex) [inline spec/classes/init_spec.rb:52]: I got hit by this a lot… Maybe we should consider increase the limit, or disable the test completely?

**Review by smortex — APPROVED** [review]
LGTM

File: spec/classes/init_spec.rb (hunk 1 above)
Comment (traylenator) [reply spec/classes/init_spec.rb:52]: Yes. 
I've tried re-arranging things to not get this elsewhere but don't quite understand how to avoid it.

**Pull request merged**